service/history/timerQueueTaskExecutorBase.go

```diff
@@ -138,11 +138,18 @@ func (t *timerQueueTaskExecutorBase) deleteWorkflow(
 	msBuilder workflow.MutableState,
 ) error {
 
-	if err := t.deleteCurrentWorkflowExecution(task); err != nil {
-		return err
-	}
-
-	if err := t.deleteWorkflowExecution(task); err != nil {
+	if err := t.shard.DeleteWorkflowExecution(
```

Comment:
I know that historically this required two calls.  But logically it shouldn't.  Why not just get rid of both these request objects and add:

shard.DeleteWorkflow(key WorkflowKey)

and let the persistence layer (who might be able to do this in one call!) sort it out?